src/displayapp/screens/Adder.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include "displayapp/apps/Apps.h"
+#include "displayapp/Controllers.h"
+#include "displayapp/screens/Screen.h"
+#include <lvgl/lvgl.h>
+#include "components/fs/FS.h"
+#include <list>
+#include <vector>
+
+namespace Pinetime {
+  namespace Applications {
+    namespace Screens {
+
+      // Save file version
+      constexpr unsigned int AdderVersion = 1;
+
+      struct AdderSave {
+        unsigned int Level {0};
+        unsigned int HighScore {0};
+        unsigned int Version {AdderVersion};
+      };
+
+      enum class AdderField { UNDEFINED, BLANK, SOLID, BODY, FOOD };
+
+      class Adder : public Screen {
+
+      public:
+        Adder(Pinetime::Components::LittleVgl& lvgl, Pinetime::Controllers::FS& fs);
+        ~Adder() override;
+
+        enum class MoveConsequence { DEATH, EAT, MOVE };
+
+        // Overridden functions
+        void Refresh() override;
+        bool OnTouchEvent(Pinetime::Applications::TouchEvents event) override;
+
+      private:
+        static constexpr const char* GameSavePath = "/games/adder/adder.sav";
+        static constexpr unsigned int TileSize = 9;
+        static constexpr unsigned int AdderDelayInterval = 200;
+
+        Pinetime::Components::LittleVgl& lvgl;
+        Controllers::FS& filesystem;
+
+        AdderSave data; // Game save data
+        AdderField* field {nullptr};
+
+        lv_task_t* refreshTask {nullptr};
+        lv_color_t* tileBuffer {nullptr};
```

Comment:
Tilebuffer is always needed. Better have it on the stack, than doing the memory management ourselfs. Also enables us to use `tileBuffer.fill()` instead of `std::fill`
```suggestion
        std::array<lv_color_t, TileSize * TileSize> tileBuffer;
```